### PR TITLE
Moving icons loading functionality to DynamoCoreWpf project

### PR DESCRIPTION
**Pull request by vmoyseenko**
#### Purpose

To remove "PresentationCore" dependency in "DynamoCore" icons loading functionality is moved to "DynamoCoreWpf" project.

#### Modifications

Created new class `IconsService` which saves links to icons warehouse for every resource assembly.

#### Additional references

[MAGN-5656](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-5656)  DUI: Rework LibraryCutomization class to remove reference on PresentationCore.dll

Unfortunately it is impossible to remove `PresentationCore` dependency as some references are presented in `Configurations.cs`.

#### Reviewers

@Benglin, please, take a look.

### CODE REVIEW

File: src/DynamoCoreWpf/Services/IconsService.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+using System.Windows;
+using System.Windows.Media.Imaging;
+using Dynamo.DSEngine;
+
+namespace Dynamo.Wpf.Services
+{
+    public class IconsService
+    {
+        private static Dictionary<Assembly, IconsWarehouse> warehouses =
+            new Dictionary<Assembly, IconsWarehouse>();
```
Comment (Benglin) [inline src/DynamoCoreWpf/Services/IconsService.cs]: Hi @vmoyseenko, I'd prefer if we do not make this `static`, can we not place an instance of `IconsService` under `DynamoViewModel`?

File: src/DynamoCoreWpf/Services/IconsService.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            var assembly = libraryCustomization.Assembly;
+
+            if (!warehouses.ContainsKey(assembly))
+                warehouses[assembly] = new IconsWarehouse(assembly);
+
+            return warehouses[assembly];
+        }
+    }
+
+    public class IconsWarehouse
```
Comment (Benglin) [inline src/DynamoCoreWpf/Services/IconsService.cs]: `IconsWarehouse`, such a creative name :)

I would have gone with `IconWarehouse` and `IconService` (both `Icon` in singular form).

I personally don't have an opinion about this, just checking in with @pboyer to see if he has a more brilliant idea?

**Comment by Benglin** [comment]
Thanks for doing this, @vmoyseenko! I'll wait for the `static` removal and then come back to it.

**Comment by vmoyseenko** [comment]
@Benglin, `statics` is removed. Please, take a look.

File: src/DynamoCore/Utilities/Configurations.cs [hunk 3]
```diff
@@ -43,7 +41,6 @@ public class Configurations
         // Grid Settings
         public static readonly int GridSpacing = 100;
         public static readonly int GridThickness = 2;
-        public static readonly Color GridLineColor = Color.FromRgb(232, 232, 232);
```
Comment (Benglin) [inline src/DynamoCore/Utilities/Configurations.cs]: Thanks for removing these redundant stuff.

**Comment by Benglin** [comment]
Thanks @vmoyseenko, LGTM!

**Pull request merged**